src/intervals/gff3/reader.jl

```diff
@@ -95,9 +95,159 @@ Return a FASTAReader initialized to parse trailing FASTA data.
 Throws an exception if there is no trailing FASTA, which can be checked using
 `hasfasta`.
 """
-function getfasta(reader::GFF3Reader)
+function getfasta(reader::Reader)
     if !hasfasta(reader)
-        error("GFF3 file has no FASTA data ")
+        error("GFF3 file has no FASTA data")
     end
     return Bio.Seq.FASTAReader(reader.state.stream)
 end
+
+info("compiling GFF3")
+const record_machine, body_machine = (function ()
+    cat = Automa.RegExp.cat
+    rep = Automa.RegExp.rep
+    rep1 = Automa.RegExp.rep1
+    alt = Automa.RegExp.alt
+    opt = Automa.RegExp.opt
```

Comment:
So I guess the purpose of this is if the machine definition is in an anonymous function, you know stuff like cat = Automa.RegExp.cat is contained only to the function, and is not causing unforseen consequences. If it's not already, we should include advice on this in the docs when telling people how to make their own parsers for new formats.